### DISCUSSION TITLE
Launcpad Pro not connecting, manual selection of Launchpad Pro Mk2 profile does nothing either

### DISCUSSION THREAD

**Discussion by FakenameMusic** [helgoboss/helgobox#1140] [General]
Hi there,
I bought a Lauchpad Pro to use with Playtime 2 but it does not auto-connect. 
I tried manually loading the supplied Launchpad Pro Profile through the main Playtime 2 UI "Controller" page, selecting the relevant "Main Profile" and the MIDI IN/OUT for the device, but this does nothing either. 
![ctrlr](https://github.com/user-attachments/assets/75c3b354-970b-48ab-a606-1ae0e6536d39)

The device works fine with Ableton and Reaper has no problem seeing / using it as a MIDI device.

Has anyone had a similar issue and may have advice?
Thanks

**Comment by helgoboss** [comment]
Video for setting up the Launchpad Pro: https://www.youtube.com/watch?v=tkyNCGOVumw

Automatic setup also takes care of configuring your REAPER MIDI device preferences correctly. For automatic setup to work, remove your manually created controller first. Also, if you have tried to select the Launchpad Pro in ReaLearn as input or output, undo that (must not be used there, otherwise it's considered an override and won't work).

Maybe you plugged it in before Playtime was running? In that case, Playtime won't bother to pick it up.

**Reply by FakenameMusic** [reply]
Thanks for your reply. Yes I had seen this video the day it was released - this is why I bought the Launchpad Pro. :)

The auto-connect does not work for me even when Playtime is running (see attached video at the end).
Although I have got the manual setup working just now, there is **one step not included in your original video** that I had to do to make it work: Go into the **Advanced Settings** and select the "Live" preset for Launchpad Pro, as this did not happen automatically - and to be honest it was only re-watching your video that I realised that **Advanced Settings** was there :)

Thank you, I will mark this issue closed, but let me know if you want to investigate more why it is not auto-connecting for me (it might help other new users) and I am glad to help. 

Video: https://www.youtube.com/watch?v=M_hCgqWUAGE

**Reply by helgoboss** [reply]
Interesting, picking the controller preset manually should actually not be necessary as it typically picks this automatically. That's why it's in the advanced settings. But maybe this didn't work since the rest of the automatic detection also didn't work.

Could you possibly do the following?

FIRST: Remove your Launchpad from the controller list again.

1. Quit REAPER
2. Disconnect your Launchpad
3. Open the Windows command line app
4. Enter the following commands (this should start REAPER while logging to a file):
    ```
    set HELGOBOX_LOG=debug
    "PATH_TO_YOUR_REAPER\REAPER.exe" > "%USERPROFILE%\Downloads\helgobox.log"
    ```
7. Start Playtime within REAPER
8. Connect your Launchpad and wait a few seconds
9. Quit REAPER
10. Reply within this issue, attaching the file "helgobox.log" in your Downloads directory (alternatively, you can also send it to info@helgoboss.org)

Thanks!

**Reply by FakenameMusic** [reply]
Here's the log: 
[helgobox.log](https://github.com/user-attachments/files/16783475/helgobox.log)
🤘🤘🤘

**Reply by helgoboss** [reply]
Thank you!

That's interesting. Somehow there's no log entry at all showing that Helgobox detected a plugged-in MIDI device. Are you sure you plugged out the Launchpad before it (sorry, didn't specifically mentioned it) and then plugged it in exactly step 6?

**Reply by helgoboss** [reply]
Apart from that, I have a slight idea why the auto-detection might have failed: The "-2 " in front of the device name. I think I will need to make the search more fuzzy. I added #1143.

Would still be nice to see the log when plugging in the device.

**Reply by FakenameMusic** [reply]
OK, good news, some progress at least:

The first time I went through the same sequence of adding the controller as you suggested before and it didn't work.
[helgobox1.log](https://github.com/user-attachments/files/16786403/helgobox1.log)

The second time I deleted the 3 Launchpad related midi inputs and the output from Reaper's MIDI IN/OUT config pages so that Reaper was starting from a blank slate of not knowing the device at all- and it worked! So perhaps the problem is not the name, but related to Reaper's way of handshaking the device when it's first connected, and how that is interacting with the autodetect in Playtime 2? In short: once I made Reaper think I was connecting the device for the first time, the auto detect in Playtime worked.
[helgobox - deleted device.log](https://github.com/user-attachments/files/16786473/helgobox.-.deleted.device.log)

Hopefully that will be of help to you.

**Reply by helgoboss** [reply]
Thanks for checking. The logs still don't contain anything useful, really weird.

But now it would be interesting to check: What's the exact name of the Launchpad Pro in the list of REAPER output devices? Maybe the "2-" prefix is gone?

**Reply by FakenameMusic** [reply]
Sorry I should have made it more explicit: the "2-" is still in the name. 
I tried changing the Device Alias in the MIDI setup page, but that made no difference.
The only thing that made the difference was to remove the device entirely from Reaper's MIDI I/O setup pages.
That's what I meant when I was suggesting that the problem was not the name.

**Reply by helgoboss** [reply]
Oh man, this info brings us back to zero. Now I have no idea what it could be.

**Reply by helgoboss** [reply]
@FakenameMusic Could I ask you for another test?

I had a look in the code and there is really **absolutely no way** this could have worked if the **output** device name is not exactly "Launchpad Pro". My assumption is that you left the controller in the list before trying it again? Or that the **output** device name changed to exactly "Launchpad Pro"?

If you don't mind, could you try the following? I tried to make the instructions a bit more accurate.

1. Disconnect your Launchpad
1. Remove your Launchpad from the controller list within Playtime (make sure it's not there anymore).
1. Make REAPER forget all Launchpads within the MIDI input **and output** device preferences
1. Close any project that you have open (we want REAPER to show a completely empty project on the next start) 
1. Quit REAPER
1. Open the Windows command line app
1. Enter the following commands (this should start REAPER while logging to a file):
    ```
    set HELGOBOX_LOG=debug
    "PATH_TO_YOUR_REAPER\REAPER.exe" > "%USERPROFILE%\Downloads\helgobox.log"
    ```
1. Start Playtime within REAPER (using the button)
1. Go to Settings => Controllers
1. Connect your Launchpad again and wait at least 20 seconds (does Playtime auto-add a controller when you do this?)
1. Delete the track containing Playtime
1. Quit REAPER
1. Reply within this issue, attaching the file "helgobox.log" in your Downloads directory (alternatively, you can also send it to info@helgoboss.org)

Thanks!

**Comment by FakenameMusic** [comment]
OK, interesting! I'm not at home at the moment, but as soon as I get back
I'll do this and make a screen cap of me doing this. It's quite possible
that yesterday, one of the times I forced reaper to forget the controller
that I missed one of the inputs or the output.  I'll get back to you a
little later.  🙌

**Comment by FakenameMusic** [comment]
OK, so long story short: auto detection didn't work until I changed the device name to take out the "2-". Once I took out the "2-" it worked.

Here is a video of the first time I tried (that it didn't work) and also the log generated:
I'm sure all you want is the log, but the video will at least perhaps let you see if I did something stupid.
They are still uploading / processing with Youtube as I type this, so if they are not visible now, or look very low quality, wait a while til YouTube does it's processing on it's end and then look again.

[helgobox-29-08 A.log](https://github.com/user-attachments/files/16797447/helgobox-29-08.A.log)
and the video:
https://youtu.be/xxxxcccccccccc

Here's the second time where the rename worked and auto connection worked:
[helgobox 29-08 B.log](https://github.com/user-attachments/files/16797482/helgobox.29-08.B.log)
https://youtu.be/ccccccccccccccvc

**Reply by helgoboss** [reply]
Thanks! That's good news. Will fix this in #1143.
